common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/ores/OreGenerator.java

```diff
@@ -82,6 +88,12 @@ private static Optional<BlockPos> computeVeinOrigin(WorldGenLevel level, ChunkGe
         ).findFirst();
     }
 
+    @Nullable
+    public GTOreDefinition getEntry(WorldGenLevel level, Holder<Biome> biome, RandomSource random) {
+        var veins = WorldGeneratorUtils.getCachedBiomeVeins(level.getLevel(), biome, random);
+        int randomEntryIndex = GTUtil.getRandomItem(random, veins, veins.size());
+        return randomEntryIndex == -1 ? null : veins.get(randomEntryIndex).getValue();
+    }
```

Comment:
Is this supposed to be here? it's not called anywhere.

Reply:
Uh i forgot it there

Reply:
```suggestion
```